linked_list/implementation.py

```diff
@@ -1,40 +1,159 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
-    def __init__(self, elements=None):
-        pass
-
+    def __init__(self, elements=None):              #Initializer, defaults with a None object
+        self.start = None                           #Set the starting node and the last one.
+        self.end = None
+        self.size = 0
+        if elements is not None:                    #Checking if there was any objects passed
+            if isinstance(elements, list):          #Checking if a list object was passed
+                if elements:                        #Checking if the list is empty
+                    self.start = Node(elements[0])  #Setting the first node as the first object in the list that was passed
+                    self.size += 1
+                    position = self.start
+                    for element in elements[1:]:    #Iterating through the list that was passed except the first object
+                        new_node = Node(element)    #Create a new node each time, and setting the position to the next node
+                        position.next = new_node
+                        position = new_node
+                        self.size += 1
+                    self.end = position
+            else:
+                new_node = Node(elements)           #If not a list, we simply create a single node with the object passed and set it as the first node
+                self.start = new_node
+                self.end = new_node
+                self.size += 1
+            
+            
     def __str__(self):
-        pass
+        if self.start is not None:                  #Checks if the linked list is empty
+            position = self.start                   #Setting position to the first node and creates a temporary list
+            list = []
+            while position is not None:             #Iterates to the end of the linked list and appends the elements of each node to the temporary list
+                list.append(position.elem)
+                position = position.next
+            return '{}'.format(str(list))
+        else:
+            return '[]'                             #Returns a represantion of an empty list if the linked list is indeed empty
+
 
     def __len__(self):
-        pass
+        return self.count()                         #Uses the count method to return the length of the linked list
+
 
     def __iter__(self):
-        pass
+        position = self.start
+        while position is not None:
+            yield position.elem
+            position = position.next
+
```

Comment:
This is a very simple and good implementation of the iterator!!! Although, would it be better to return the "Node" instead of elem, seeing that the "fields" in LinkedList are Nodetype